api/src/main/java/org/pmiops/workbench/cdr/CdrVersionService.java

```diff
@@ -97,27 +97,69 @@ public DbCdrVersion findAndSetCdrVersion(Long cdrVersionId) {
     return dbCdrVersion;
   }
 
-  /**
-   * Retrieve all the CDR versions visible to users with the specified data access level. When
-   * {@link DataAccessLevel#PROTECTED} is provided, CDR versions for both {@link
-   * DataAccessLevel#REGISTERED} and {@link DataAccessLevel#PROTECTED} are returned. Note: this
-   * relies on {@link DbUser#dataAccessLevel} accurately reflecting that the user is in the
-   * authorization domain that has access to the CDR version BigQuery data sets with the matching
-   * {@link DataAccessLevel} values.
-   *
-   * @param dataAccessLevel the data access level of the user
-   * @return a list of {@link DbCdrVersion} in descending timestamp, data access level order.
-   */
-  public List<DbCdrVersion> findAuthorizedCdrVersions(DataAccessLevel dataAccessLevel) {
-    ImmutableSet<Short> visibleValues = DATA_ACCESS_LEVEL_TO_VISIBLE_VALUES.get(dataAccessLevel);
-    if (visibleValues == null) {
-      return ImmutableList.of();
+  public Optional<DbCdrVersion> findByCdrVersionId(Long cdrVersionId) {
+    return Optional.ofNullable(cdrVersionDao.findByCdrVersionId(cdrVersionId));
+  }
+
+  @Deprecated // only handles the Registered Tier
+  public CdrVersionListResponse getCdrVersions() {
+    DataAccessLevel accessLevel = userProvider.get().getDataAccessLevelEnum();
+    if (accessLevel == DataAccessLevel.REGISTERED) {
+      CdrVersionMapResponseInner registeredTierVersions =
+          getVersionsForTier(accessTierService.getRegisteredTier());
+      return new CdrVersionListResponse()
+          .items(registeredTierVersions.getVersions())
+          .defaultCdrVersionId(String.valueOf(registeredTierVersions.getDefaultCdrVersionId()));
+    } else {
+      throw new ForbiddenException("User does not have access to any CDR versions");
     }
-    return cdrVersionDao.findByDataAccessLevelInOrderByCreationTimeDescDataAccessLevelDesc(
-        visibleValues);
   }
 
-  public Optional<DbCdrVersion> findByCdrVersionId(Long cdrVersionId) {
-    return Optional.ofNullable(cdrVersionDao.findByCdrVersionId(cdrVersionId));
+  public CdrVersionMapResponse getCdrVersionsByTier() {
+    List<DbAccessTier> tiers = accessTierService.getAccessTiersForUser(userProvider.get());
+    if (tiers.isEmpty()) {
+      throw new ForbiddenException("User does not have access to any CDR versions");
+    }
+
+    final CdrVersionMapResponse response = new CdrVersionMapResponse();
```

Comment:
optional: I would probably combine these 3 lines into one return

Reply:
The generated CdrVersionMapResponse doesn't provide an interface that allows that.

this is possible but I'd rather not:
`return (CdrVersionMapResponse) tiers.stream().map(this::getVersionsForTier).collect(Collectors.toList())`

Reply:
Ah, I didn't realize that this response was actually an array, rather than an object containing an array field. I might suggest using an object response instead to allow the API model to evolve in the future (without breaking changes), i.e. if you ever wanted to add a top level field to the response.